### PR TITLE
feat: test verifying CRUD project operations on KG API

### PR DESCRIPTION
**Pull request by jachro**
This PR updates graph to `2.42.0` and adds a new test verifying that CRUD operations can be done with the new KG APIs.

/deploy

### CODE REVIEW

**Comment by olevski** [comment]
@jachro can this be an integration tests in the KG repo?

**Comment by jachro** [comment]
We were thinking about that but it wouldn't be easy and not that robust as it requires many services to be reachable. The other thing is, as this is the public Renku API, it seems reasonable to test that on the acceptance level against any deployment of choice. It actually executes quite fast and there should be no issues with flakiness as it interacts just with APIs. I mean, for now, it still uses a browser to log in, but in the future hopefully, that won't be necessary and we'll log in via the API.

File: CHANGELOG.rst [hunk 1]
```diff
@@ -1,5 +1,38 @@
 .. _changelog:
 
+0.37.0
+------
+
+Renku ``0.37.0`` introduces new Project Creation and updated Project Update API.
```
Comment (bethcg) [inline CHANGELOG.rst:6]: How about ... improves the Knowledge Graph API, with a new Project Creation functionality and a Project Update enhancement.

**Review by bethcg — DISMISSED** [review]

**Review by bethcg — APPROVED** [review]

**Comment by jachro** [comment]
> Hey @jachro - thanks for providing the link to the swagger content.
> 
> I have a couple of q's:
> 
> * is this introducing a new major version of the API? I see you somehow have major and minor versions in the swagger endpoint

KG API is not versioned at the moment. Where did you see the version?

> * is there some documentation of this API somewhere, which could perhaps be pointed to in the release notes?

I think the best would be to link the swagger page from Renkulab.io but unfortunately, this can be done only after the release is deployed. The other thing we could do is to add a link to [the relevant section in the README](https://github.com/SwissDataScienceCenter/renku-graph/tree/2.42.0/knowledge-graph#post-knowledge-graphprojects) of the knowledge-graph service. Not sure, if that's better, though.

**Comment by seanrmurphy** [comment]
> > Hey @jachro - thanks for providing the link to the swagger content.
> > I have a couple of q's:
> > 
> > * is this introducing a new major version of the API? I see you somehow have major and minor versions in the swagger endpoint
> 
> KG API is not versioned at the moment. Where did you see the version?

Ah - when I asked for the swagger link, you sent me the link to the core service and not the KG; this confused me.

The versions I saw were the core service versions.

> > * is there some documentation of this API somewhere, which could perhaps be pointed to in the release notes?
> 
> I think the best would be to link the swagger page from Renkulab.io but unfortunately, this can be done only after the release is deployed. The other thing we could do is to add a link to [the relevant section in the README](https://github.com/SwissDataScienceCenter/renku-graph/tree/2.42.0/knowledge-graph#post-knowledge-graphprojects) of the knowledge-graph service. Not sure, if that's better, though.

I just think it would be good if there are new endpoint functionalities to point to a place where a user can get more info if interested. I think it would be good to add a link to the docs.

**Review by seanrmurphy — APPROVED** [review]
Thanks for making the changes @jachro .

**Comment by jachro** [comment]
> Thanks for making the changes @jachro .

Thanks for your review @seanrmurphy!

**Review by bethcg — APPROVED** [review]

**Comment by olevski** [comment]
@jachro it would be nice if @eikek or someone else who is versed in Scala reviews this. I dont know any Scala so I cannot review the code.

**Review by aledegano — APPROVED** [review]

**Comment by jachro** [comment]
> @jachro it would be nice if @eikek or someone else who is versed in Scala reviews this. I dont know any Scala so I cannot review the code.

Hi @olevski. Yes, I'll ask Eike to take a look but, although it may look like quite a significant change, it's just a clean-up after the recent tests removal plus an addition to the Project Create/Update API spec. We'll be making some more changes to the tests soon as we'd like to stop using a browser in the old tests.

For now, the most important change is the `CHANGELOG`.

**Pull request merged**